### DISCUSSION TITLE
How can I make alignment in table with some table styles?

### DISCUSSION THREAD

**Discussion by hwiorn** [kjambunathan/org-mode-ox-odt#116] [Q&A]
I tried like below. But I can't make alignment rows differently.

#### A sample of table
I want to set center the first row and specific alignment others.

```org
** Document version history
#+ATTR_ODT: :widths "15, 30, 100, 25"
#+ATTR_ODT: :list-table t
- | <c1> |   <c2>    | <l8>       |  <c1>  |
-
  - Version
  - Date
  - Description
  - Author
- ----------------
  - 1.0
  - 2021-10-27
  - Initial document
  - Tim Lee
- ----------------
  - 1.1
  - 2021-10-28
  -
    - asasasasa
    - aoeuuu
  - Tim Lee
- ----------------
  - 1.1
  - 2021-10-28
  -
    - asasasasa
    - aoeuuu
  - Tim Lee
- ----------------
  - 1.1
  - 2021-10-28
  -
    - asasasasa
    - aoeuuu
  - Tim Lee

```
![image](https://user-images.githubusercontent.com/1394254/139361067-7feb0629-f1c0-4c39-8433-c3af21c24e58.png)

#### A sample of table
I want to set center the first row and specific alignment others.

```org
#+ATTR_ODT: :style "GriddedTable"
#+ATTR_ODT: :list-table t
- | <l1> | <l2> | <c2> | < c2> |
-
  - Dataset
  - Metric
  - Data amount
  - Result
- ----------------
  - Ib1
  - WER(%)[fn:wer]
  - 361112.1 (sec.)
  - 25.1%
- ----------------
  - ClovalCall
  - WER(%)[fn:wer]
  - 1234.5 (sec.)
  - 20.1%
```
![image](https://user-images.githubusercontent.com/1394254/139361112-c76c0d7b-10fc-42b8-b6ff-0f582ec0cb6d.png)

**Comment by QiangF** [comment]
It seems that the org table alignment specification does not with with both list table and table with column span.

    #+macro: s1 string
    #+attr_odt: :style "GriddedTable"
    #+attr_odt: :span "@1$1{3:1} @1$2{1:8} @2$2{1:2} @2$4{1:2} @2$6{1:2} @2$8{1:2}"
    |   <c2>   | <r2>  | <c2> | <c2> | <c2> | <c2> | <c2> | <c2> | <c2> |
    | {{{s1}}} | Sales |      |      |      |      |      |      |      |
    |          |  Q1   |      |  Q2  |      |  Q3  |      |  Q4  |      |
    |          |  foo  | bar  | foo  | bar  | foo  | bar  | foo  | bar  |
    |  North   |  350  |  46  | 253  |  34  | 234  |  42  | 382  |  68  |
    |  South   |  462  |  84  | 511  |  78  | 435  |  45  | 534  |  89  |

**Reply by kjambunathan** [reply]
> It seems that the org table alignment specification does not with with both list table and table with column span.

This is the intended behaviour.  Without this workaround, there will be a stackoverflow because of recursive calls.

This is the https://github.com/kjambunathan/org-mode-ox-odt/issues/25.  You made a request to  Nicolas Goaziou, but he refused to look at this bug, saying that the ODT exporter was not part of the core.  It doesn't matter.  I will remove this limitation, atleast partially, in the next few days.  See https://github.com/kjambunathan/org-mode-ox-odt/blob/e90550e5df70e6385eb3f70733ba4644cbd9948d/lisp/ox-odt.el#L5767

**Comment by hwiorn** [comment]
> I will remove this limitation, atleast partially, in the next few days.

I'm willing to wait this update.

**Comment by kjambunathan** [comment]
> I want to set center the first row and specific alignment others.

I have siphoned off this part of the request as [When using Custom Table styles it would be useful to honor alignment cookies](https://github.com/kjambunathan/org-mode-ox-odt/issues/122).

FWIW, the purpose of the `org-odt-table-styles` is roughly equivalent to applying autoformat on tables in LibreOffice.  Please track the above bug report.

**Comment by kjambunathan** [comment]
> #+ATTR_ODT: :style "GriddedTable"
> #+ATTR_ODT: :list-table t
> - | <l1> | <l2> | <c2> | < c2> |
> -
>   - Dataset
>   - Metric
>   - Data amount
>   - Result
> - ----------------
>   - Ib1
>   - WER(%)[fn:wer]
>   - 361112.1 (sec.)
>   - 25.1%
> - ----------------
>   - ClovalCall
>   - WER(%)[fn:wer]
>   - 1234.5 (sec.)
>   - 20.1%


**_You don't have either paragraph content or "really long" content.  So, you really don't need a list table for it._**  If some descriptions are long, and not paragraph-like, you can either add a shrinking cookie to the column or you can siphon it off as a footnote reference and make it a transcluded table.

**Reply by hwiorn** [reply]
> You don't have either paragraph content or "really long" content. So, you really don't need a list table for it. If some descriptions are long, and not paragraph-like, you can either add a shrinking cookie to the column or you can siphon it off as a footnote reference and make it a transcluded table.

Yeah, you're right.

At that time, I didn't know the transcluded table and I thought about reading test results from a CSV(TSV) file and write to a ODT file as a report document to automate them all.
So I thought the list table is a suitable format for it.

**Comment by kjambunathan** [comment]
> I want to set center the first row and specific alignment others.

A quick heads up ... I have added some basic set of changes as part of https://github.com/kjambunathan/org-mode-ox-odt/commit/3221803c7e98ee09a3433841a63a90b340f31ed0.  The changes too longer than expected, and I ended up overhauling and enhancing table-related code.  I may have to fix some loose ends, before you can take it for test run.

I will add a note on usage as soon as I do a second pass.  Please wait for further instructions ....
